### ISSUE TITLE
Thunderbird 115.5.1

### ISSUE THREAD

**Issue by critt74** [TB-throwback/LookOut-fix-version#112]
Hi,
I recently update to Thunderbird 115.5.1 (from 102.15.1) and I update Lookout to 6.1 version. Unfortunately the file winmail.dat is not unpacked and the attachments are not visible. Remove Winmail.dat is not set. 
![Cattura](https://github.com/TB-throwback/LookOut-fix-version/assets/152958857/0114dcca-1044-48af-a8f2-214ebb742692)
Thank you

**Comment by dugite-code** [comment]
Hi, please open the Error console (Ctrl+Shift+J) and look for any errors or warnings.

Also is the option Strict Content Type Matching enabled?

**Comment by critt74** [comment]
Yes there are errors
![Cattura](https://github.com/TB-throwback/LookOut-fix-version/assets/152958857/c2ded664-927d-4b61-b71d-f82bce471b79)

and yes, it is enabled
thank you
Cristina

**Comment by critt74** [comment]
Hi,
my folder is not corrupt, I'll explain my problem better. I receive file .eml (externally Thunderbird) from certified e-mail address system (PEC in Italy) and I open this file with Thunderbird. Some time ago with thunderbird 102 and Lookout everything worked fine, I opened the eml file and if winmail.dat was there it was opened, but now it isn't. So today I try to import file .eml in a Thunderbird folder and....it works!!
While a single file .eml don't run ?

**Comment by dugite-code** [comment]
Ah I see the issue now. When you open an exported email saved to disk on your computer (in .eml format) in Thunderbird it fails to decode the winmail.dat file.

When you import that email back into the Thunderbird folder structure it will be able to decode.

# Technical explanation
[L163](https://github.com/TB-throwback/LookOut-fix-version/blob/master/src/api/Attachment/implementation.js#L163) is using msgHdr.folder to fetch the msg URI. As the email isn't in a folder this will always return `null`

**Comment by dugite-code** [comment]
The uri not being defined would potentially:

- Cause issues with other plugins
- Prevent sending the attachments in a decoded format

As we currently don't support that simply returning and empty string when msgHdr.folder.getUriForMsg(msgHdr) is null might adress the issue. Needs a bit of further investigation.

**Comment by bixgit** [comment]
Is there a fix to this bug?
@critt74 did you find a solution?

**Comment by critt74** [comment]
> Is there a fix to this bug? [@critt74](https://github.com/critt74) did you find a solution?

No, i don't find a solution. Luckily I have few cases now. When this happens I use a program external to Thunderbird, the name is Winmail Opener
